pages/auth.tsx

```diff
@@ -15,54 +43,71 @@ const debug = debugFactory('abacus:pages/auth.tsx')
  */
 const AuthPage = function AuthPage() {
   debug('AuthPage#render')
-  const [error, setError] = useState<null | string>(null)
+  const classes = useStyles()
+
+  const [error, setError] = useState<null | AuthError>(null)
   useEffect(() => {
-    if (typeof window !== 'undefined') {
-      // If we have the hash with the authorization info, then extract the info, save
-      // it for later, and go to the "home" page.
-      //
-      // The hash should look something like the following upon success:
-      // #access_token=...&expires_in=#######&scope=global&site_id=0&token_type=bearer
-      // The hash should look something like the following upon failure:
-      // #error=access_denied&error_description=You+need+to+log+in+to+WordPress.com&state=
-      if (window.location.hash && window.location.hash.length > 1) {
-        const authInfo = qs.parse(window.location.hash.substring(1))
-        if (authInfo.access_token && authInfo.scope === 'global' && authInfo.token_type === 'bearer') {
-          const expiresInSeconds = toInt(authInfo.expires_in)
-          const experimentsAuthInfo = {
-            accessToken: authInfo.access_token as string,
-            expiresAt: typeof expiresInSeconds === 'number' ? Date.now() + expiresInSeconds * 1000 : null,
-            scope: 'global',
-            type: 'bearer',
-          }
-          saveExperimentsAuthInfo(experimentsAuthInfo)
+    // This is needed because of server-side rendering
+    if (typeof window === 'undefined') {
+      console.warn('onExperimentAuthCallbackUrl: Could not find `window`')
+      return
+    }
 
-          window.location.replace(window.location.origin)
-        } else if (authInfo.error === 'access_denied') {
-          setError('Please log into WordPress.com and authorize Abacus - Testing to have access.')
+    if (!window.location.hash || window.location.hash.length === 0) {
+      console.error('Authentication Error:', 'Missing hash in auth callback url.')
+      setError(AuthError.UnknownError)
```

Comment:
String errors are more descriptive and easier to debug:

```suggestion
      setError('Authentication Error: Missing hash in auth callback url.')
      return
```

Also, we can't expect all our users to be comfortable with the console, so if we include all the information as a string, it'll be easier to debug based on screenshots.